### ISSUE TITLE
Add haversine cdist

### ISSUE THREAD

**Issue by MuellerSeb** [mapado/haversine#30]
Hey there,

nice package! I was wondering, if you could implement a routine to compute a pairwise distance matrix like [`scipy.spatial.distance.cdist `](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html#scipy.spatial.distance.cdist).

Cheers,
Sebastian

**Comment by MuellerSeb** [comment]
Related implementation: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.haversine_distances.html

**Comment by jdeniau** [comment]
Hi,

Thanks !

If your issue is with performance, #26 added the `haversine_vector` function to optimize performance.

You can see how to use it [in the documentation](https://github.com/mapado/haversine#performance-optimisation-for-distances-between-all-points-in-two-vectors)

If that's not performance-related, can you explain why you need this ?

**Comment by MuellerSeb** [comment]
cdist is constructing a n times m matrix for two input arrays with n and m points respectively. The vector version does not do this.

**Comment by jdeniau** [comment]
OK so let me rephrase to make sure I understand your needs : 

You have a list of geopoints, and you need to calculate the distance between every points :

pseudo-code :

```py
LYON = (45.7597, 4.8422)
PARIS = (48.8567, 2.3508)
NEW_YORK = (40.7033962, -74.2351462)

calculate_each_distances([LYON, PARIS, NEW_YORK])
```
and the output should be something like that :
```
[
  [    0,  392, 6163],
  [  392,    0, 5853],
  [ 6163, 5853,    0],
]
```

Is that it ?

**Comment by MuellerSeb** [comment]
Almost. This is the [pdist](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html#scipy.spatial.distance.pdist) functionality.

One step further is, to provide two lists of points for rows and cols in the resulting matrix.

**Comment by Noezor** [comment]
This already exists, see https://github.com/mapado/haversine/blob/0b47ac417d82c4ae15ba49c7856453bf81afa1b9/tests/test_haversine_vector.py#L21-L30

However, I believe [the `comb` parameter](https://github.com/mapado/haversine/blob/3e61c151aeddd0520fa26eab695da0ccc3b42c45/haversine/haversine.py#L105) is not clear from the docstring (one needs to read the implementation). 
@jdeniau do you mind if I rework the docstring a bit?

**Comment by jdeniau** [comment]
@Noezor help yourself! Every improvement in the documentation is helpful 👍
